common/src/main/java/tachyon/security/PlainSaslServer.java

```diff
@@ -76,29 +75,36 @@ public String getMechanismName() {
         throw new IllegalArgumentException("Invalid message format, parts must contain 3 items");
       }
       String authzid = parts[0];
-      mAuthcid = parts[1];
+      String authcid = parts[1];
       String passwd = parts[2];
-      if (mAuthcid == null || mAuthcid.isEmpty()) {
+      if (authcid == null || authcid.isEmpty()) {
         throw new IllegalStateException("No authentication identity provided");
       }
       if (passwd == null || passwd.isEmpty()) {
         throw new IllegalStateException("No password provided");
       }
       if (authzid == null || authzid.isEmpty()) { // authzid = authcid
-        authzid = mAuthcid;
+        authzid = authcid;
+      } else if (!authzid.equals(authcid)) {
+        // TODO: support impersonation
+        throw new UnsupportedOperationException("Impersonation is not supported now.");
       }
 
       NameCallback nameCallback = new NameCallback("User");
-      nameCallback.setName(mAuthcid);
+      nameCallback.setName(authcid);
       PasswordCallback passwordCallback = new PasswordCallback("Password", false);
       passwordCallback.setPassword(passwd.toCharArray());
-      AuthorizeCallback authCallback = new AuthorizeCallback(mAuthcid, authzid);
+      AuthorizeCallback authCallback = new AuthorizeCallback(authcid, authzid);
 
       Callback[] cbList = {nameCallback, passwordCallback, authCallback};
       mHandler.handle(cbList);
       if (!authCallback.isAuthorized()) {
         throw new SaslException("AuthorizeCallback authorized failure");
       }
+      mAuthzid = authCallback.getAuthorizedID();
+
+      // After verification succeeds, a user with this authz id will be set to a Threadlocal.
+      RemoteClientUser.set(mAuthzid);
```

Comment:
Question about using this Threadlocal storage: it seems we only `set` and `remove`, where do we really use this thread local variable?